### DISCUSSION TITLE
Improved "offroad" rendering for unpaved roads.

### DISCUSSION THREAD

**Discussion by julcnx** [osmandapp/OsmAnd#12476] [Ideas]
As an active OSM contributor and off-road enthusiast, I love OSMAnd and the live maps updates feature which allows me to do some quick updates before a ground survey.

One major pain I have with the offroad rendering is that surface is not taken into account.
e.g. a paved track will be shown like an unpaved track, an unpaved tertiary road will look like a paved one...

I am currently improving the minor road network in northern Thailand, and one part is to update existing highway classifications and add a surface tag when missing. Because of rendering limitations, there has been a bad habit of tagging unpaved roads as track without a surface, even if these roads are in fact tertiary, unclassified, residential, service...

My proposal would be to either:
- improve existing offroad renderer to incorporate surface tag information (e.g. using dashed or straight casing/stroke lines)
- create a new separate style to achieve this.

I already found a few resources online:
- code related to each rendering style e.g. offroad is here https://github.com/osmandapp/OsmAnd-resources/blob/master/rendering_styles/offroad.render.xml
- documentation https://docs.osmand.net/ru/main@latest/development/osmand-file-formats/osmand-rendering-style

What I am not finding is instructions on how to test rendering on my local machine.

PS: my goal is not to build my own Android application, but to have an improved or separate offroad style in the official APP so that other users can use it.

Any pointers would be much appreciated. Thanks!

**Comment by vshcherb** [comment]
Hello @cmoffroad, offroad style was created by one of volunteers. So I will copy that information to him but that is not guaranteed he will manage to add surface information. You always can try to create you own style. 
Probably testing on your mobile device will be easier unless you can fingure out how to run OsmAndMapCreator with Offline Rendering (works only on Linux now).

**Reply by julcnx** [reply]
@vshcherb Great, I would be also happy to create and submit a new style (targeting enduro riders). 
What would be the process for this? I assume I would need to develop and test a new style e.g. enduro.render.xml and submit a pull request?

**Update:** I have downloaded the latest PBF from my country, converted it to OBF for a small area directly inside OsmAndMapCreator (Mac). I cannot figure out next how to render/generate tiles based on a given XML file inside the working directory (default.render.xml). Any pointers would be much appreciated :)

**Reply by vshcherb** [reply]
Yes, if you create a rendering style we will share it across community, though we only include rendering styles that authors are ready to use & maintain long term. 

Creating rendering style is anyway quite tricky... If you call it enduro.render.xml that could lead to many requests related to support specific motorcycle issue, so you are welcome :-)

**Reply by julcnx** [reply]
@vshcherb I am happy to develop and maintain a new style if I can come up with something useful.

I just need to figure out first a proper development/testing workflow. You might not have seen my update in the comment above. Do you have any pointers/articles on the missing step to make the offline rendering work with OsmAndMapCreator?

**Reply by vshcherb** [reply]
No this part of documentation is not complete and might not be complete (cause it works only with Linux for now), you can ask some questions in Telegram chat, so it could go faster to setup.
